qubes-rpc/qvm-template-repo-query

```diff
@@ -49,16 +56,38 @@ touch -r "$hashfile" "$repodir/template.repo"
 RET=0
 
 if [ "$1" = "query" ]; then
-    dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{reponame}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|' "$SPEC"
+    if $DNF5; then
+        dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{repoid}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|\n' "$SPEC"
+    else
+        dnf repoquery "${OPTS[@]}" --qf='%{name}|%{epoch}|%{version}|%{release}|%{repoid}|%{downloadsize}|%{buildtime}|%{license}|%{url}|%{summary}|%{description}|' "$SPEC"
+    fi
     RET="$?"
 elif [ "$1" = "download" ]; then
     # Download/retry algorithm: take mirrors in random order. In this order,
     # try to download from the first one - if download failed but anything was
     # downloaded - retry from the same one. If download failed and nothing was
     # downloaded, go to the next one. The intention is to retry on interrupted
     # connection, but skip mirrors that are not synchronized yet.
-    urls="$(dnf downloadurl "${OPTS[@]}" --url --all-mirrors "$SPEC" | shuf)"
-    readarray -t urls <<<"$urls"
+    declare -a urls=()
```

Comment:
Does it make sens if you assign plain string to `urls` below anyway? And then `readarray` will create array later. Only the middle case benefits from this declare, but all cases should be consistent.

Reply:
Then I guess the 1st case should use something else for temporary storage of space-separated urls. I will amend it and push the change in a second.